File: src/main/scala/info/crlog/higgs/Event.scala
package info.crlog.higgs

/**
 * @author Courtney Robinson <courtney@crlog.info>
 */
object Event extends Enumeration {
  /**
   * Any event which this handler provides notifications for
   */
  type Event = Value
  val CHANNEL_ACTIVE = Value("channel_active")
  val CHANNEL_INACTIVE = Value("channel_inactive")
  val CHANNEL_REGISTERED = Value("channel_registered")
  val CHANNEL_UNREGISTERED = Value("channel_unregistered")
  val EXCEPTION_CAUGHT = Value("exception_caught")
}

File: events/src/main/java/io/higgs/events/Event.java
package io.higgs.events;

import java.util.Arrays;

/**
 * @author Courtney Robinson <courtney@crlog.info>
 */
public class Event {
    private final Object[] params;
    private final String name;

    public Event(String event, Object[] params) {
        this.name = event;
        this.params = params;
    }

    /**
     * @return the name of this event
     */
    public String name() {
        return name;
    }

    /**
     * @return the set of arguments emitted with this event
     */
    public Object[] params() {
        return params;
    }

    @Override
    public String toString() {
        return "Event{" +
                "name='" + name + '\'' +
                ", params=" + Arrays.toString(params) +
                '}';
    }
}
